src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1078 @@
+              return IOUtils.toString(timeLimitProcess.getErrorStream(), Charset.defaultCharset());
+            } catch (IOException e) {
+              // Error reading from process's error stream.
+              return "Error reading from process's error stream.";
+            }
+          }
+        };
+
+    ExecutorService fixedThreadPool = Executors.newFixedThreadPool(2);
+    Future<String> stdOutput = fixedThreadPool.submit(stdCallable);
+    Future<String> errOutput = fixedThreadPool.submit(errCallable);
+    fixedThreadPool.shutdown();
+
+    // Wait for the TimeLimitProcess to finish.
+    try {
+      timeLimitProcess.waitFor();
+    } catch (InterruptedException e) {
+      return new Outputs("", "Process was interrupted while waiting.", 1);
+    }
+    if (timeLimitProcess.timed_out()) {
+      timeLimitProcess.destroy();
+      return new Outputs("", "Process timed out after " + timeoutLimit * 1000 + " seconds.", 1);
+    }
+
+    String stdOutputString;
+    String errOutputString;
+    try {
+      stdOutputString = stdOutput.get();
+      errOutputString = errOutput.get();
+    } catch (InterruptedException e) {
+      return new Outputs("", "Process was interrupted while waiting.", 1);
+    } catch (ExecutionException e) {
+      return new Outputs("", "A computation in the process threw an exception.", 1);
+    }
+
+    // Collect and return the results from the standard output and error output.
+    return new Outputs(stdOutputString, errOutputString, timeLimitProcess.exitValue());
+  }
+
+  /**
+   * Normalize the standard output obtained from running a JUnit test suite. By normalizing the
+   * {@code String} representation of the output, we remove any extraneous information such as line
+   * numbers. The relevant information that is retained includes the indices of each failing test
+   * and the respective method names.
+   *
+   * @param input the {@code String} produced from running a JUnit test suite
+   * @return a {@code String} which contains several {@code String} pairs of output from the JUnit
+   *     test suite. Each pair's first element is the index of the failing test along with the name
+   *     of the method containing the failing assertion. Each pair's second element is the output
+   *     from the failing assertion. For example: {@code 1) test10(ErrorTestLangMinimized)
+   *     java.lang.AssertionError: Contract failed: compareTo-equals on fraction1 and fraction4}
+   */
+  private static String normalizeJUnitOutput(String input) {
+    Scanner scn = new Scanner(input);
+
+    // JUnit output starts with index 1 for first failure
+    int index = 1;
+    // String to represent the result that we return
+    String res = "";
+    // Continue until we finish processing the input String
+    while (scn.hasNextLine()) {
+      String line = scn.nextLine();
+      // Look for a String of the form "i)" where i is the current failing
+      // test index.
+      if (line.startsWith(index + ") ")) {
```

Comment:
The minimizer now retains the whole stack trace. Previously I had thought that only the method name and name of the failure were necessary. I had also needed a way to handle the line numbers and the initial approach ignored all lines with line numbers.